### PR TITLE
Test gem with real Rails app instead of stubbing

### PR DESCRIPTION
**Pull request by kirs**
This gem provides integration with Rails and Sprockets, but the test suite tries to [mock Rails](https://github.com/fphilipe/premailer-rails/blob/bdba2041b9da892cea69927340ef8da9b34f5b34/spec/support/stubs/rails.rb) instead of using an actual dummy Rails app.

This PR adds a dummy app to the test suite. It allows us to get rid of Rails stubs, and write an full integration test which checks that the gem hooks into ActionMailer.

Using dummy app in the test suite is a very common pattern. It's basically used by any gem that integrates with Rails, for instance [Devise](https://github.com/plataformatec/devise/tree/master/test/rails_app). Stubbing Rails is not the best way to test that the gem works correctly with Rails.

### CODE REVIEW

File: spec/integration/delivery_spec.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+describe 'ActionMailer::Base delivery' do
+  it 'delivers email with inlined CSS' do
+    WelcomeMailer.welcome_email("world").deliver_now
+
+    mail = ActionMailer::Base.deliveries.last
+    expect(mail).to be_present
+    body = mail.html_part.body.to_s
+    expect(body).to be_present
+    expect(body).to include(%{<p style="font-size: 12px;">Hello world</p>})
```
Comment (kirs) [inline spec/integration/delivery_spec.rb]: This is the new integration test that checks if hooking into ActionMailer works.

File: spec/support/stubs/action_mailer.rb [hunk 2]
```diff
@@ -1,5 +0,0 @@
-module ActionMailer
-  class Base
-    def self.register_interceptor(x); end
```
Comment (kirs) [inline spec/support/stubs/action_mailer.rb]: No more ActionMailer stubs.

**Comment by kirs** [comment]
@artygus let me know how this PR looks for you!

File: spec/dummy/app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    <%= csrf_meta_tags %>
+
+    <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+    <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
+  </head>
+
+  <body>
+    <%= yield %>
+  </body>
+</html>
```
Comment (fphilipe) [inline spec/dummy/app/views/layouts/application.html.erb]: Is this file needed?

File: spec/dummy/app/views/welcome_mailer/welcome_email.html.erb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+<link rel="stylesheet" href="/assets/application.css"/>
```
Comment (fphilipe) [inline spec/dummy/app/views/welcome_mailer/welcome_email.html.erb]: Can we move this to the layout?

File: spec/dummy/app/assets/javascripts/application.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+// Any JavaScript/Coffee file within this directory, lib/assets/javascripts, vendor/assets/javascripts,
+// or any plugin's vendor/assets/javascripts directory can be referenced here using a relative path.
+//
+// It's not advisable to add code directly here, but if you do, it'll appear at the bottom of the
+// compiled file. JavaScript code in this file should be added after the last require_* statement.
+//
+// Read Sprockets README (https://github.com/rails/sprockets#sprockets-directives) for details
+// about supported directives.
+//
+//= require_tree .
```
Comment (fphilipe) [inline spec/dummy/app/assets/javascripts/application.js]: I guess we don't really need this file.

File: spec/dummy/config/initializers/application_controller_renderer.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+# Be sure to restart your server when you modify this file.
+
+# ApplicationController.renderer.defaults.merge!(
+#   http_host: 'example.org',
+#   https: false
+# )
```
Comment (fphilipe) [inline spec/dummy/config/initializers/application_controller_renderer.rb]: I'd be fine with leaving out all comment only files that aren't directly necessary for the dummy app in order to reduce the noise and just have the very essential needed to make this work. I've done it that way in another gem of mine: https://github.com/fphilipe/warden-github-rails/tree/master/spec/rails_app

File: spec/integration/delivery_spec.rb (hunk 1 above)
Comment (fphilipe) [reply spec/integration/delivery_spec.rb]: 👍

File: spec/dummy/app/views/welcome_mailer/welcome_email.html.erb (hunk 4 above)
Comment (kirs) [reply spec/dummy/app/views/welcome_mailer/welcome_email.html.erb]: I don't think there's a layout for emails? The purpose of this mailer view is to test inlining (see `spec/integration/delivery_spec.rb`)

Comment (fphilipe) [reply spec/dummy/app/views/welcome_mailer/welcome_email.html.erb]: Isn't this view rendered inside `spec/dummy/app/views/layouts/mailer.html.erb`?

**Comment by kirs** [comment]
Addressed all your comments 👌

**Comment by fphilipe** [comment]
Landed in c9a3b783f459ec4f5044a125f7fa79d411e5013e. Thanks! Would you mind rebasing the other PRs?
